packages/rum-core/src/domain/assembly.ts

```diff
@@ -78,8 +78,15 @@ export function startRumAssembly(
           ;(serverRumEvent.usr as RumEvent['usr']) = commonContext.user as User & Context
         }
 
+        let beforeSendResult
         if (configuration.beforeSend) {
-          limitModification(serverRumEvent, FIELDS_WITH_SENSITIVE_DATA, configuration.beforeSend)
+          beforeSendResult = limitModification(serverRumEvent, FIELDS_WITH_SENSITIVE_DATA, configuration.beforeSend)
+        }
+        if (beforeSendResult === false && serverRumEvent.type !== RumEventType.VIEW) {
+          return
+        }
+        if (beforeSendResult === false) {
+          console.warn(`Can't dismiss view events using beforeSend, use onNewLocation instead!`)
```

Comment:
Not sure why you put the beforeSendResult checks outside of the configuration.beforeSend if.
If the intention was to avoid to introduce another if level, what about extracting the before send logic into a function?